aerospike_discovery.py

```diff
@@ -488,54 +575,172 @@ def _authenticate(self, user, password, password_field_id):
             sys.exit(STATE_UNKNOWN)
     except Exception as e:
         print("Failed to authenticate connection to the Aerospike cluster at %s:%s"%(args.host,args.port))
-        print e
+        print(e)
+        sys.exit(STATE_UNKNOWN)
+
+asinfo_cmd = "statistics"
+
+# Find server version to find correct asinfo commands.
+req = 'build'
+try:
+    version = client.info(req).split('.')
+except Exception as e:
+        print("Failed to execute asinfo command %s on the Aerospike cluster at %s:%s"%(req, args.host, args.port))
+        print(e)
         sys.exit(STATE_UNKNOWN)
 
+use_latencies_cmd = True if int(version[0]) >= 5 and int(version[1]) >= 1 else False
+latency_histogram = ''
+latency_time = ''
+
+if args.dc:
+        asinfo_cmd='dc/'+args.dc
+
+        # Version 5.0+ got removed dc/DC_NAME and added get-stats:context=xdr;dc=DC_NAME
+        if int(version[0]) >= 5:
+            asinfo_cmd='get-stats:context=xdr;dc='+args.dc
+
+# namespace must be checked after sets, bins, and sindex
+elif args.set:
+    asinfo_cmd='sets/'+args.namespace+'/'+args.set
```

Comment:
Please give proper spacing.

Reply:
I think you mean like `asinfo_cmd = 'sets/' + args.namespace + '/' + args.set`? I thought it looked weird as well